core/dbt/context/providers.py

```diff
@@ -1262,6 +1262,10 @@ def submit_python_job(self, parsed_model: Dict, compiled_code: str) -> AdapterRe
             )
         return self.adapter.submit_python_job(parsed_model, compiled_code)
 
+    @contextproperty
+    def macros(self) -> Dict[str, Any]:
+        return self.manifest.macros
```

Comment:
Should the return type hint be changed to `MutableMapping[str, ParsedMacro]`
or does each macro needs to be converted to dict like
`return {k: v.to_dict() for k, v in self.manifest.macros.items()}`

Reply:
Yes, if we were to do this, we'd want to turn `manifest.macros` into a read-only dictionary, akin to what's done in [`build_flat_graph`](https://github.com/dbt-labs/dbt-core/blob/56783446dbc765d93c43afad2b763302f034ef6b/core/dbt/contracts/graph/manifest.py#L690-L701).